### PR TITLE
Basic instructions on building and starting a local environment

### PR DESCRIPTION
**Pull request by bittrance**
Some instructions on how to get started with development.

### CODE REVIEW

**Comment by bittrance** [comment]
Currently, on most cases (but not all?) following this procedure results in one node failing to start with this error while running the end-to-end tests work. Not sure why; seems to be on a random node.

```json
{
    "level": "error",
    "ts": 1714147847.252328,
    "caller": "leaderelection/leaderelection.go:332",
    "msg": "error retrieving resource lock spegel/spegel-leader-election: Get \"https://10.96.0.1:443/apis/coordination.k8s.io/v1/namespaces/spegel/leases/spegel-leader-election\": dial tcp 10.96.0.1:443: i/o timeout\n",
    "stacktrace": "k8s.io/client-go/tools/leaderelection.(*LeaderElector).tryAcquireOrRenew\n\t/go/pkg/mod/k8s.io/client-go@v0.28.8/tools/leaderelection/leaderelection.go:332\nk8s.io/client-go/tools/leaderelection.(*LeaderElector).acquire.func1\n\t/go/pkg/mod/k8s.io/client-go@v0.28.8/tools/leaderelection/leaderelection.go:252\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\t/go/pkg/mod/k8s.io/apimachinery@v0.28.8/pkg/util/wait/backoff.go:226\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\t/go/pkg/mod/k8s.io/apimachinery@v0.28.8/pkg/util/wait/backoff.go:227\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/go/pkg/mod/k8s.io/apimachinery@v0.28.8/pkg/util/wait/backoff.go:204\nk8s.io/client-go/tools/leaderelection.(*LeaderElector).acquire\n\t/go/pkg/mod/k8s.io/client-go@v0.28.8/tools/leaderelection/leaderelection.go:251\nk8s.io/client-go/tools/leaderelection.(*LeaderElector).Run\n\t/go/pkg/mod/k8s.io/client-go@v0.28.8/tools/leaderelection/leaderelection.go:208\nk8s.io/client-go/tools/leaderelection.RunOrDie\n\t/go/pkg/mod/k8s.io/client-go@v0.28.8/tools/leaderelection/leaderelection.go:228\ngithub.com/spegel-org/spegel/pkg/routing.(*KubernetesBootstrapper).Run\n\t/build/pkg/routing/bootstrap.go:84\ngithub.com/spegel-org/spegel/pkg/routing.(*P2PRouter).Run\n\t/build/pkg/routing/p2p.go:127\nmain.registryCommand.func3\n\t/build/main.go:173\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\t/go/pkg/mod/golang.org/x/sync@v0.7.0/errgroup/errgroup.go:78"
}
```

**Comment by phillebaba** [comment]
I have already created a contributing documentation. Could you see if you can merge your additional information to that document instead?

https://github.com/spegel-org/spegel/blob/main/CONTRIBUTING.md

**Comment by bittrance** [comment]
Putting build instructions with contribution guidelines indeed seems to be common in Kubernetes-land. TIL. Mildly embarrassing - had I read the contribution guidelines, I would have seen those instructions... Anyway, I moved the instructions and adjusted somewhat to match the style.

**Comment by bittrance** [comment]
I also noted that the image build command example sets both `IMG` and `TAG` which the Makefile does not really support (since it only folds `TAG` into `IMG` when using default value). I fixed the command so it works, but perhaps we should really fix the Makefile?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -39,9 +39,38 @@ make docker-build
 It is possible to specify a different image name and tag.
 
 ```shell
-make docker-build IMG=exmaple.com/spegel TAG=feature
+make docker-build IMG=example.com/spegel:feature
```
Comment (phillebaba) [inline CONTRIBUTING.md:42]: Leave this as it is and lets fix the Makefile in a separate PR.

Comment (bittrance) [reply CONTRIBUTING.md]: Done.

**Pull request merged**